integration-tests/load/ccip/destination_gun.go

```diff
@@ -0,0 +1,146 @@
+package ccip
+
+import (
+	"context"
+	"fmt"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog"
+	"github.com/smartcontractkit/chainlink-testing-framework/wasp"
+	"github.com/smartcontractkit/chainlink/deployment"
+	ccipdeployment "github.com/smartcontractkit/chainlink/deployment/ccip"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/utils"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/router"
+	"sync/atomic"
+	"time"
+)
+
+type ChainSelectorPair struct {
+	src uint64
+	dst uint64
+}
+
+type DestinationGun struct {
+	l             zerolog.Logger
+	env           deployment.Environment
+	seqNums       map[ChainSelectorPair]*atomic.Uint64
+	roundNum      *atomic.Int32
+	chainSelector uint64
+	receiver      common.Address
+	loki          *wasp.LokiClient
+}
+
+func NewDestinationGun(l zerolog.Logger, chainSelector uint64, env deployment.Environment, receiver common.Address, loki *wasp.LokiClient) *DestinationGun {
+	seqNums := make(map[ChainSelectorPair]*atomic.Uint64)
+	for _, cs := range env.AllChainSelectorsExcluding([]uint64{chainSelector}) {
+
+		seqNums[ChainSelectorPair{
+			src: cs,
+			dst: chainSelector,
+		}] = &atomic.Uint64{}
+		seqNums[ChainSelectorPair{
+			src: chainSelector,
+			dst: cs,
+		}].Store(1)
+	}
+	return &DestinationGun{
+		l:             l,
+		env:           env,
+		seqNums:       seqNums,
+		roundNum:      &atomic.Int32{},
+		chainSelector: chainSelector,
+		receiver:      receiver,
+		loki:          loki,
+	}
+}
+
+func (m *DestinationGun) Call(_ *wasp.Generator) *wasp.Response {
+	m.roundNum.Add(1)
+	requestedRound := m.roundNum.Load()
+
+	waspGroup := fmt.Sprintf("%d-%s", m.chainSelector, "messageOnly")
+
+	state, err := ccipdeployment.LoadOnchainState(m.env)
+	if err != nil {
+		return &wasp.Response{Error: err.Error(), Group: waspGroup, Failed: true}
+	}
+
+	src, err := m.MustSourceChain()
+	if err != nil {
+		return &wasp.Response{Error: err.Error(), Group: waspGroup, Failed: true}
+	}
+
+	csPair := ChainSelectorPair{
+		src: src,
+		dst: m.chainSelector,
+	}
+	m.seqNums[csPair].Add(1)
+	m.l.Info().
+		Int32("RoundNum", requestedRound).
+		Uint64("Destination ChainSelector", m.chainSelector).
+		Uint64("Source ChainSelector", src).
+		Uint64("SequenceNumber", m.seqNums[csPair].Load()).
+		Msg("starting transmit")
+
+	r := state.Chains[src].Router
+
+	msg, err := m.GetMessage()
+	if err != nil {
+		return &wasp.Response{Error: err.Error(), Group: waspGroup, Failed: true}
+	}
+
+	fee, err := r.GetFee(
+		&bind.CallOpts{Context: context.Background()}, src, msg)
+	if err != nil {
+		return &wasp.Response{Error: err.Error(), Group: waspGroup, Failed: true}
```

Comment:
it will help if you use deployment.MayBeDataError(err) here, it will show you the revert reason, if there is any